include/dem/adaptive_sparse_contacts.h

```diff
@@ -0,0 +1,522 @@
+   * next DEM time steps.
+   *
+   * @param particle_handler The particle handler that contains all the particles
+   *
+   * @param g The external forces, same for all cells (gravity or other sources)
+   *
+   * @param force The contact or hydrodynamic forces
+   *
+   * @param dt The DEM time step
+   */
+  void
+  update_average_velocities_acceleration(
+    Particles::ParticleHandler<dim> &particle_handler,
+    const Tensor<1, 3>              &g,
+    const std::vector<Tensor<1, 3>> &force,
+    const double                     dt);
+
+  /**
+   * @brief Carrie out the identification of the mobility status of each cell
+   * through a node-based identification and check. Only the active and ghost
+   * cells are processed.
+   *
+   * The following 4 checks (search loops) are done:
+   *
+   * 1. Check if the cell is empty (n_particle = 0), if so, nodes and cells are
+   * flagged as empty mobility status (5)
+   *
+   * 2. Check if the cell is mobile by criteria (average granular temperature >
+   * threshold, solid fraction < threshold or has at least one empty node from
+   * previous check), if so, nodes are flagged and cells are stored with mobile
+   * mobility status (4)
+   *
+   * 3. Check if the cell is mobile by neighbor (at least a node is flagged as
+   * mobile from previous check), if so, cells are stored in map as mobile
+   * status (4) and nodes that are not mobile are flagged as active (1/3)
+   *
+   * 4. Check if the cell is active (at least a node is flagged as active from
+   * previous check), if so, cells are stored with active status in the map
+   * (1/3)
+   *
+   * The remaining cells are inactive (0)
+   *
+   * @param background_dh The dof handler of the background grid
+   *
+   * @param particle_handler The particle handler that contains all the particles
+   *
+   * @param n_active_cells The number of active cells in triangulation
+   *
+   * @param mpi_communicator The MPI communicator
+   */
+  void
+  identify_mobility_status(
+    const DoFHandler<dim>                 &background_dh,
+    const Particles::ParticleHandler<dim> &particle_handler,
+    const unsigned int                     n_active_cells,
+    MPI_Comm                               mpi_communicator);
+
+  /**
+   * @brief Carrie out the identification of the mobility status of each cell
+   * through a node-based identification and check. Only the active and ghost
+   * cells are processed. For CFD-DEM, the node status have to be reset at each
+   * first DEM iteration of a CFD time step as mobile in order to update
+   * all forces.
+   *
+   * @param background_dh The dof handler of the background grid
+   *
+   * @param particle_handler The particle handler that contains all the particles
+   *
+   * @param n_active_cells The number of active cells in triangulation
+   *
+   * @param mpi_communicator The MPI communicator
+   *
+   * @param counter The counter of the DEM iteration
+   */
+  void
```

Comment:
This is a small function, I would inline it.

Reply:
Even if it called a big function?

Reply:
yeah because the call itself won't be inlined. it will be point to the actual implementation of the function without anyway issue :)